File: embeded-c/Day14/demo04.c
#include<stdio.h>

typedef enum type{
	BAT, BALL, ALL
}ptype_t;

typedef struct{
	char name[20];
	int age;
	ptype_t type;
	int matches;
}player_t;

char *to_string(ptype_t type)
{
	switch(type)
	{
		case BAT:	return "Batsman";
		case BALL:	return "Bolwer";
		case ALL:	return "All Rounder";
	}
	return NULL;
}


void print_player(player_t *p)
{
	printf("Player : %s %d %s %d\n", p->name, p->age, to_string(p->type), p->matches);
}


int main(void)
{
	player_t arr[3] = {
			{"abc", 28, BAT, 122},
			{"xyz", 26, BALL, 150},
			{"pqr", 29, ALL, 170}
		};

	//arr[i] - player of ith index - player_t
	
	for(int i = 0 ; i < 3 ; i++)
		print_player(&arr[i]);

	return 0;
}

File: iot/Day06/demo04.py

def function1():
    t1 = (
        (1, 2, 3, 4),
        (10, 20, 30, 40),
        (11, 22, 33, 44)
    )

    #print(t1)

    #for t in t1:
    #    print(t)

    for t in t1:
        for e in t:
            print(e, end=" ")
        print("")

#function1()

def function2():
    students = (
        (2, "abc", 97.0),
        (9, "xyz", 89.5),
        (15, "pqr", 91.7)
    )

    #print(students)

    #for stud in students:
    #    print(stud)
    
    #for stud in students:
    #    print(f"rollno = {stud[0]}, name = {stud[1]}, marks = {stud[2]}")

    for rollno, name, marks in students:
        print(f"rollno = {rollno}, name = {name}, marks = {marks}")

function2()